Lesson16/ecenaz/script.js

```diff
@@ -0,0 +1,490 @@
+    - Log the count.
+*/
+function countOccurrences(arr,word){
+
+  let count= 0;
+  for (let i=0; i < arr.length; i++){
+if(arr[i]===word){
+  count++
+}
+  }
+console.log(count);
+}
+countOccurrences(["elma", "armut", "ananas", "elma", "elma","muz"],"elma");
+/*
+19. Remove Falsy Values
+    - Define a function `removeFalsyValues(arr)` that loops through an array
+      and returns a new array without falsy values (false, 0, "", null, undefined, NaN).
+    - Log the new array.
+*/
+function removeFalsyValues(arr){
+  let newArray=[];
+  for (let i=0; i<arr.length; i++){
+    if(arr[i]){
+      newArray.push(arr[i]);
+    }
+  }
+  console.log(newArray);
+}
+removeFalsyValues([false, 0, "numara","eşya","saat", "", null, undefined, NaN]);
+/*
+20. Sum of All Digits in a String
+    - Define a function `sumDigits(str)` that loops through each character of `str`,
+      checks if it's a digit, and if so, adds it to a total sum.
+    - Log the final sum.
+    - Example: "abc123" -> 6
+*/
+function sumDigits(str){
+  let sum=0;
+  for(let i=0; i<str.length; i++){
+    if(str[i]>= '0' && str[i]<='9'){
+      sum += parseInt(str[i]);
+    }
+  }
+  console.log(sum);
+}
+sumDigits("abc123");
+/*
+21. Average of Array Elements
+    - Define a function `averageArray(numbers)` that uses a loop
+      to calculate the average (sum / length).
+    - Log the average.
+*/
+function averageArray(numbers){
+  let sum=0;
+  for (let i=0; i<numbers.length; i++ ){
+sum += numbers[i];
+  }
+  let averageArray = sum / numbers.length;
+  console.log(averageArray);
+}
+averageArray([5,10,15,20,25,30]);
+/*
+22. Flatten a 2D Array (Nested Loops)
+    - Define a function `flattenArray(twoDArray)` that takes an array of arrays
+      (e.g., [[1,2],[3,4]]) and uses nested loops to create a new one-dimensional array.
+    - Log the flattened array.
+*/
+function flattenArray(twoDArray){
+  let flattenedArray= [];
+  for (let i=0; i<twoDArray.length; i++){
+    for (let j=0; j<twoDArray.length; j++){
+      flattenedArray.push(twoDArray[i][j]);
+    }
+  }
+  console.log(flattenedArray);
+}
+
+flattenArray([[1,2],[3,4]]);
+/*
+23. Find Words Containing a Letter
+    - Define a function `findWordsWithLetter(words, letter)` that loops through
+      an array of words and returns a new array of only the words that contain
+      the given letter.
+    - Log the filtered array.
+*/
+function findWordsWithLetter(words, letter){
+  let filteredArray=[];
+for (let i=0; i<words.length; i++){
+  if(words[i].includes(letter)){
+    filteredArray.push(words[i]);
+  }
+}
+ console.log(filteredArray);
+}
+findWordsWithLetter(["elma", "armut","ananas","avokado","muz","turp"],"a");
+
+/*
+24. Push and Pop Operations
+    - Define a function `pushPopExample(arr, itemToPush)` that:
+      - pushes itemToPush to arr
+      - logs the updated array
+      - then pops the last element
+      - logs the popped element
+      - logs the final array
+*/
+function pushPopExample(arr, itemToPush){
```

Comment:
In this function, the expectation is that itemToPush is a single item, not an array. Therefore, you don't need to use loops.